From 30cc6389b2d40a02a720b7de0df1041ec389cafe Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 31 Dec 2012 01:25:46 -0500
Subject: [PATCH] EMC Anvil Tweaks

Make Items not have a repair cost increase due to Anvil usage.
Renaming is only 5 levels, combines include free renames.
---
 src/main/java/net/minecraft/server/ContainerAnvil.java | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/main/java/net/minecraft/server/ContainerAnvil.java b/src/main/java/net/minecraft/server/ContainerAnvil.java
index 9b73555..df71dac 100644
--- a/src/main/java/net/minecraft/server/ContainerAnvil.java
+++ b/src/main/java/net/minecraft/server/ContainerAnvil.java
@@ -1,6 +1,7 @@
 package net.minecraft.server;
 
 import java.util.Iterator;
+import java.util.List;
 import java.util.Map;
 
 import org.apache.commons.lang3.StringUtils;
@@ -271,8 +272,11 @@ public class ContainerAnvil extends Container {
                 this.a = 39;
             }
 
+            if (j == i && j > 0) { this.a = Math.min(15, 4 + itemstack.count); } // EMC - renames scale 5 to 15 levels
+            else if (j > 0) { this.a -= j; } // EMC - Free renames on combining
+
             if (this.a >= 40 && !this.n.abilities.canInstantlyBuild) {
-                itemstack1 = null;
+                this.a = 39; // EMC - cap at 39
             }
 
             if (itemstack1 != null) {
@@ -299,6 +303,16 @@ public class ContainerAnvil extends Container {
         }
     }
 
+    // EMC Start - send modified max level on change
+    @Override
+    public void b() {
+        super.b();
+        for (ICrafting listener : (List<ICrafting>) this.listeners) {
+            listener.setContainerData(this, 0, this.a);
+        }
+    }
+    // EMC end
+
     public void addSlotListener(ICrafting icrafting) {
         super.addSlotListener(icrafting);
         icrafting.setContainerData(this, 0, this.a);
-- 
1.8.4.4

